### ISSUE TITLE
MRZ mul test failing on riscv-xcrypto architecture.

### ISSUE THREAD

**Issue by ben-marshall** [scarv/libscarv#4]
The MRZ test, when compiled for `riscv-xcrypto` and run through Spike, results in a "User load segfault", indicating that there is some memory corruption occuring.

```sh
$> make ARCH=riscv-xcrypto run-tests
...
$> tail -n 20 build/riscv-xcrypto/work/test_mrz.out 
  print 'r == %s' % ( hex( r ) )                         
  print '  != %s' % ( hex( t ) )                         
  sys.exit(1)                                            
                                                         
# mrz mul 8 / 10                                         
z  00000000 ra 00010214 sp 7fbe8a80 gp 0001e848          
tp 00000000 t0 000122d8 t1 0000000f t2 4e9ade62          
s0 00000000 s1 00000000 a0 0001e048 a1 7fbe8b84          
a2 00000004 a3 00000009 a4 00000000 a5 00000001          
a6 00000800 a7 00000400 s2 00000000 s3 00000024          
s4 7fbe8b84 s5 00000009 s6 0001e048 s7 00000004          
s8 00000000 s9 00000000 sA 00000000 sB 00000000          
t3 8c11f623 t4 0b584892 t5 abc2d3eb t6 cd1b2686          
pc 0001152c va 0000000c insn       ffffffff sr 80046020  
User load segfault @ 0x0000000c                          
```

**Comment by ben-marshall** [comment]
- This bug is transient, it only occurs for some executions and not others.
- It turns out there is no ASM code involved, I managed to make the bug happen when targeting `riscv` as well.

**Comment by danpage** [comment]
Fixed in new v1.0.0 branch: it's was more a problem of the test infrastructure than the kernel.
